xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/actionbuttons/actionButtons.js

```diff
@@ -96,14 +96,14 @@ var XWiki = (function(XWiki) {
 
     addShortcuts: function() {
       var shortcuts = {
-        'action_cancel' : "$services.localization.render('core.shortcuts.edit.cancel')",
-        'action_preview' : "$services.localization.render('core.shortcuts.edit.preview')",
+        'action_cancel' : "$escapetool.javascript("#getShortcutValue('core.shortcuts.edit.cancel')")",
+        'action_preview' : "$escapetool.javascript("#getShortcutValue('core.shortcuts.edit.preview')")",
         // The following 2 are both "Back to edit" in the preview mode, depending on the used editor
-        'action_edit' : "$services.localization.render('core.shortcuts.edit.backtoedit')",
-        'action_inline' : "$services.localization.render('core.shortcuts.edit.backtoedit')",
-        'action_save' : "$services.localization.render('core.shortcuts.edit.saveandview')",
-        'action_propupdate' : "$services.localization.render('core.shortcuts.edit.saveandview')",
-        'action_saveandcontinue' : "$services.localization.render('core.shortcuts.edit.saveandcontinue')"
+        'action_edit' : "$escapetool.javascript("#getShortcutValue('core.shortcuts.edit.backtoedit')")",
+        'action_inline' : "$escapetool.javascript("#getShortcutValue('core.shortcuts.edit.backtoedit')")",
+        'action_save' : "$escapetool.javascript("#getShortcutValue('core.shortcuts.edit.save')")",
+        'action_propupdate' : "$escapetool.javascript("#getShortcutValue('core.shortcuts.edit.saveandview')")",
+        'action_saveandcontinue' : "$escapetool.javascript("#getShortcutValue('core.shortcuts.edit.saveandcontinue')")"
```

Comment:
Just as a suggestion, I think it would be much easier to read if this used `$escapetool.json()` to serialize one big map from Velocity to a JSON object.

Reply:
Somehow I just noticed when testing that the crtUser is not properly set when in the context of velocity evaluation of this script. The shortcuts are not properly set here since we use `$crtUserDoc`... Looking further into it.

My guess is that we need to make sure `frequentlyUserDoc.vm` is loaded even for this velocity template evaluation. I need to find the place responsible for evaluating this .js file.

Reply:
Addressed the main topic of this comment and the bug I found in https://github.com/xwiki/xwiki-platform/pull/2981/commits/150f1e6fc7948400cffb459ae1910e484b34151a :+1: 

Note that the solution to the bug is the low level solution. It's not a solution to the fact that the frequentlyUsedDocs template is not in the context of `$xwiki.jsfx.use`.

Reply:
Seems like this error I made is related to https://jira.xwiki.org/browse/XWIKI-19455